### PR TITLE
Simplify span handling, add infrastructure inheritence

### PR DESCRIPTION
**Pull request by Jim3Things**
This adds a wrapped span that is used to inherit the infrastructure attribute on spans, it adds a non-WithSpan sequence, modifies users.go & DBUsers.go to use that, with some cleanup of the http error handling.

The new start span, and the existing trace event methods, are moved to a common tracing level.  The client and server subdirectories will move to contain client and service interceptors, as the WithSpan variants are removed.

### CODE REVIEW

File: internal/services/frontend/DBUsers.go [hunk 1]
```diff
@@ -31,62 +32,66 @@ type DBUsers struct {
...
-					"CloudChamber: standard %q account is not using using configured password - error %v",
-					cfg.WebServer.SystemAccount, err)
-			}
-
-			return nil
+		if err = bcrypt.CompareHashAndPassword(
+			existingUser.GetPasswordHash(),
+			[]byte(cfg.WebServer.SystemAccountPassword)); err != nil {
+			tracing.Infof(
+				ctx, tick,
```
Comment (MikeCJohnsonUk) [inline internal/services/frontend/DBUsers.go:88]: Do I remember hearing you saying something about storing the current "tick" value in the ctx or am I getting mixed up with something else?

Is that something we could do, letting the trace code just extract the value and then update the ctx value at the appropriate time? 

Just a notion, not even a suggestion.

File: internal/tracing/spanEx.go [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+	"go.opentelemetry.io/otel/api/global"
+	"go.opentelemetry.io/otel/api/kv"
+	"go.opentelemetry.io/otel/api/trace"
+
+	"github.com/Jim3Things/CloudChamber/pkg/protos/log"
+)
+
+// spanEx extends the OpenTelemetry Span structure with additional features
+// needed by CloudChamber
+type spanEx struct {
```
Comment (MikeCJohnsonUk) [inline internal/tracing/spanEx.go:18]: Does this portend spanExEx and/or spaceEx2?

Sorry, couldn't resist ;-)

File: internal/services/frontend/DBUsers.go (hunk 1 above)
Comment (Jim3Things) [reply internal/services/frontend/DBUsers.go:88]: That's what I think we'll end up doing -- the discussion about storing the tick in the context is a future item though.

File: internal/tracing/spanEx.go (hunk 2 above)
Comment (Jim3Things) [reply internal/tracing/spanEx.go:18]: Open to other names.  Just couldn't think of one that wouldn't be confusing or really long...

File: internal/services/frontend/DBUsers.go (hunk 1 above)
Comment (MikeCJohnsonUk) [reply internal/services/frontend/DBUsers.go:88]: Fair enough

File: internal/tracing/spanEx.go (hunk 2 above)
Comment (MikeCJohnsonUk) [reply internal/tracing/spanEx.go:18]: Just yanking your chain right now. I've no suggestions.

**Review by MikeCJohnsonUk — APPROVED** [review]

**Pull request merged**